src/ir.cc

```diff
@@ -181,7 +181,7 @@ void LocalTypes::Set(const TypeVector& types) {
   Type type = types[0];
   Index count = 1;
   for (Index i = 1; i < types.size(); ++i) {
-    if (types[i] != type) {
+    if (!types[i].IsSame(type)) {
```

Comment:
The name might not be the best. Anyway, the purpose is that `==` only compares the enum, while the index is also important for references.